### ISSUE TITLE
[android] MyLocationView perspective change animation missing

### ISSUE THREAD

**Issue by LukasPaczos** [mapbox/mapbox-gl-native#9163]
**Platform:** Android
**Mapbox SDK version:** v5.1.0-beta.3

![tilt_issue_2](https://cloud.githubusercontent.com/assets/16925074/26683391/e7cead60-46e3-11e7-9df9-ba3b0d07e659.gif)

Animation is missing when animating camera like this:
```
CameraPosition position = map.getCameraPosition();
CameraPosition.Builder builder = new CameraPosition.Builder(position);
if (tilted) {
    tilted = false;
    builder.tilt(0);
}
else {
    builder.tilt(60);
    tilted = true;
}
map.animateCamera(CameraUpdateFactory.newCameraPosition(builder.build()));
```

Animation works fine when changing perspective via gesture, but there is another issue: #9161.
As well as #9161, this issue was not present on v4.2.0. 
In this case, map padding does not matter.

**Comment by tobrun** [comment]
Fix for this in #9174

**Comment by tobrun** [comment]
Fix has landed on release branch, closing.
